src/mono/mono/mini/debugger-agent.c

```diff
@@ -6849,6 +6870,16 @@ vm_commands (int command, int id, guint8 *p, guint8 *end, Buffer *buf)
 	case CMD_VM_STOP_BUFFERING:
 		/* Handled in the main loop */
 		break;
+	case MDBGPROT_CMD_VM_READ_MEMORY: {
+		guint8* memory = (guint8*) decode_long (p, &p, end);
+		int size = decode_int (p, &p, end);
+		buffer_add_byte_array (buf, memory, size);
+		break;
+	}
+	case MDBGPROT_CMD_VM_SET_USING_ICORDBG: {
+		agent_config.using_icordbg = TRUE;
```

Comment:
I don't like this for some reason, but I don't know why.  I see it's used to change some of the protocol encoding.  Can't we encode this in the protocol version in some way?  (Or will that break the old sdb-based debugging?)

(if that's not reasonable, we could extend CMD_VM_SET_PROTOCOL_VERSION to include a "client is icordbg" flag if the major/minor is new enough.  and then send "false" from debugger-libs)